### PR TITLE
`DropdownMenu` can be expanded to its parent size

### LINKED ISSUE #125199: [Material 3] `DropdownMenu` not expanding according to its parent's width.
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Create a `DropdownMenu` widget inside a widget.

### Expected results

`DropdownMenu`'s width should be same as its parent.

### Actual results

`DropdownMenu`'s width is small and not according to its parent's width.

### Code sample

<details><summary>Code sample</summary>

```dart
return Padding(
  padding: const EdgeInsets.symmetric(vertical: 8),
  child: DropdownMenu(
    label: const Text("Select Establishment"),
    inputDecorationTheme: InputDecorationTheme(
      filled: Theme.of(context).brightness == Brightness.dark,
      border: Theme.of(context).brightness == Brightness.dark ? null : const OutlineInputBorder()
    ),
    dropdownMenuEntries: []
  ),
);
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

Screenshot_2023-04-20-18-44-59-10_8910c767e4d7264453bba01d16c4effa

</details>

### Logs

<details><summary>Logs</summary>

```console
No logs
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.7.11, on Ubuntu 22.10 5.19.0-40-generic, locale en_PH.UTF-8)
    • Flutter version 3.7.11 on channel stable at /home/maverick/SDKs/Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f72efea43c (9 days ago), 2023-04-11 11:57:21 -0700
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/maverick/SDKs/Android
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = /home/maverick/SDKs/Android
    • ANDROID_SDK_ROOT = /home/maverick/SDKs/Android
    • Java binary at: /home/maverick/SDKs/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 15.0.7
    • cmake version 3.24.2
    • ninja version 1.11.0
    • pkg-config version 0.29.2

[✓] Android Studio (version 2022.1)
    • Android Studio at /home/maverick/SDKs/android-studio
    • Flutter plugin version 72.1.2
    • Dart plugin version 221.6096
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.2)
    • IntelliJ at /home/maverick/IDEs/idea-IU-213.6777.52
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.77.3)
    • VS Code at /usr/share/code
    • Flutter extension version 3.62.0

[✓] Connected device (3 available)
    • RMX2195 (mobile) • 192.168.1.101:40507 • android-arm64  • Android 11 (API 30)
    • Linux (desktop)  • linux               • linux-x64      • Ubuntu 22.10 5.19.0-40-generic
    • Chrome (web)     • chrome              • web-javascript • Google Chrome 112.0.5615.165

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by QuncCccccc**
Fixes #125199

This PR is to add a new property `expandedInsets` so that the `DropdownMenu` can be expandable and has some margins around.

<details><summary>Example: Setting `expandedInsets` to `EdgeInsets.zero`</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const DropdownMenuExample());

class DropdownMenuExample extends StatefulWidget {
  const DropdownMenuExample({super.key});

  @override
  State<DropdownMenuExample> createState() => _DropdownMenuExampleState();
}

class _DropdownMenuExampleState extends State<DropdownMenuExample> {
  final TextEditingController colorController = TextEditingController();
  ColorLabel? selectedColor;

  @override
  Widget build(BuildContext context) {
    final List<DropdownMenuEntry<ColorLabel>> colorEntries = <DropdownMenuEntry<ColorLabel>>[];
    for (final ColorLabel color in ColorLabel.values) {
      colorEntries.add(
        DropdownMenuEntry<ColorLabel>(value: color, label: color.label, enabled: color.label != 'Grey'),
      );
    }

    return MaterialApp(
      theme: ThemeData(
        useMaterial3: true,
        colorSchemeSeed: Colors.green,
      ),
      home: Scaffold(
        body: Center(
          child: Container(
            width: 500,
            height: 500,
            color: Colors.orange,
            child: DropdownMenu<ColorLabel>(
              expandedInsets: EdgeInsets.zero,
              inputDecorationTheme: const InputDecorationTheme(
                filled: true,
                fillColor: Colors.white,
                border: OutlineInputBorder(),
              ),
              controller: colorController,
              dropdownMenuEntries: colorEntries,
              onSelected: (ColorLabel? color) {
                setState(() {
                  selectedColor = color;
                });
              },
              // expandedInsets: EdgeInsets.only(left: 35.0, right: 20.0, top: 80),
            ),
          ),
        ),
      ),
    );
  }
}

enum ColorLabel {
  blue('Blue', Colors.blue),
  pink('Pink', Colors.pink),
  green('Green', Colors.green),
  yellow('Yellow', Colors.yellow),
  grey('Grey', Colors.grey);

  const ColorLabel(this.label, this.color);
  final String label;
  final Color color;
}
```

Screenshot 2023-06-28 at 11 33 20 PM

</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 1]
```diff
@@ -277,6 +278,17 @@ class DropdownMenu<T> extends StatefulWidget {
   /// contain space for padding.
   final List<DropdownMenuEntry<T>> dropdownMenuEntries;
 
+  /// Enables the text field to expand and provides margins.
+  ///
+  /// If this is null, the width of the text field will be determined by the width
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:283]:
```suggestion
  /// If this property is null, the width of the text field will be determined by the width
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 2]
```diff
@@ -277,6 +278,17 @@ class DropdownMenu<T> extends StatefulWidget {
...
+  /// of menu items or [DropdownMenu.width]. If this is not null, the text field
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:284]:
```suggestion
  /// of menu items or [DropdownMenu.width]. If this property is not null, the text field
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 3]
```diff
@@ -277,6 +278,17 @@ class DropdownMenu<T> extends StatefulWidget {
...
+  /// will be expanded to its parent size. The values of [expandedInsets] are
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:285]:
```suggestion
  /// field's width will match the parent's width plus the specified insets. The values of [expandedInsets] are
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 4]
```diff
@@ -277,6 +278,17 @@ class DropdownMenu<T> extends StatefulWidget {
...
+  /// margins for the text field. In other words, If this is [EdgeInsets.zero],
+  /// the width of the text field will be the same as its parent's width.
+  ///
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:288]: Just to make sure that it's clear; could add one more statement:

The [expandedInsets]' top and bottom are ignored, only its left and right properties are used.

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 5]
```diff
@@ -549,6 +561,108 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
...
+          // on its parent width. So we don't need to use `_DropdownMenuBody` to
+          // calculate the children's width.
+          return textField;
+        }
+
+        return _DropdownMenuBody(
+          width: widget.width,
+          children: <Widget>[
+            textField,
+            for (final Widget c in _initialMenu!) c,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:650]: Presumably `c` is a menu "item". So maybe use `item` instead of just `c`

**Review by HansMuller — COMMENTED** [review]
Just some feedback about the API doc

File: packages/flutter/lib/src/material/dropdown_menu.dart (hunk 4 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/dropdown_menu.dart:293]: I see. Thanks for helping review this! Just updated!

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 6]
```diff
@@ -277,6 +278,22 @@ class DropdownMenu<T> extends StatefulWidget {
...
+  /// plus the horizontal width of the specified insets.
+  ///
+  /// If this property is null, the width of the text field will be determined
+  /// by the width of menu items or [DropdownMenu.width]. If this property is not null,
+  /// the text field's width will match the parent's width plus the specified insets.
+  /// The values of [expandedInsets] are margins for the text field. In other words,
+  /// If this is [EdgeInsets.zero], the width of the text field will be the same
+  /// as its parent's width.
+  ///
+  /// The [expandedInsets]'s top and bottom are ignored, only its left and right
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:291]:
```suggestion
  /// The [expandedInsets]' top and bottom are ignored, only its left and right
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 7]
```diff
@@ -277,6 +278,22 @@ class DropdownMenu<T> extends StatefulWidget {
...
   final List<DropdownMenuEntry<T>> dropdownMenuEntries;
 
+  /// Defines the menu text field's width to be equal to its parent's width
+  /// plus the horizontal width of the specified insets.
+  ///
+  /// If this property is null, the width of the text field will be determined
+  /// by the width of menu items or [DropdownMenu.width]. If this property is not null,
+  /// the text field's width will match the parent's width plus the specified insets.
+  /// The values of [expandedInsets] are margins for the text field. In other words,
+  /// If this is [EdgeInsets.zero], the width of the text field will be the same
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown_menu.dart:288]:
```suggestion
  /// If the value of this property is [EdgeInsets.zero], the width of the text field will be the same
```

**Review by HansMuller — APPROVED** [review]
I made a few more tweaks to the API doc

**Pull request merged**